### PR TITLE
Fix failing initContainer incase of classicFullStack and metadataEnrichment combination

### PR DESCRIPTION
**Pull request by luhi-DT**
Ticket: https://dt-rnd.atlassian.net/browse/K8S-10259

## Description

Since we want to combine classicFullStack and metaDataEnrichment, combining those two will cause an error -> the agent mounts over /var/lib/dynatrace/enrichment, causing the initContainer to fail with createFile errors (lack of permission) on the hostFileSystem.
We should instead of directly creating the files there, create them in a temp location and mount the with subPath (only files not the whole folder) 
Afterwards the agent will make modifications to not overwrite our files

## How can this be tested?

Apply a Dynakube with metadata enrichment only and check if all files (.json and .properties are created)
Apply another Dynakube afterwards with classicFullStack and metadataEnrichment to check if the containers start successfully (note: the files will still be gone, but the agent has a ticket to resolve this)

### CODE REVIEW

File: pkg/consts/enrichment_injection.go [hunk 1]
```diff
@@ -2,12 +2,14 @@ package consts
...
 	EnrichmentInjectedEnv        = "METADATA_ENRICHMENT_INJECTED"
 	EnrichmentWorkloadKindEnv    = "DT_WORKLOAD_KIND"
 	EnrichmentWorkloadNameEnv    = "DT_WORKLOAD_NAME"
+	EnrichmentPropertiesFilename = "dt_metadata.properties"
+	EnrichmentJsonFilename       = "dt_metadata.json"
 )
 
 var (
 	EnrichmentMountPath = "/var/lib/dynatrace/enrichment"
+	EnrichmentInitPath  = "/tmp/enrichment"
```
Comment (0sewa0) [inline pkg/consts/enrichment_injection.go:14]: why aren't these `const`?

Comment (luhi-DT) [reply pkg/consts/enrichment_injection.go:14]: Not sure, did not want to change them, but IMHO they should be consts, good point

Comment (luhi-DT) [reply pkg/consts/enrichment_injection.go]: done

**Review by 0sewa0 — APPROVED** [review]
tested it, deployed sample apps didn't have problems starting

**Review by waodim — APPROVED** [review]
tested it, got no problems applying sample apps when reapplied a classicfullstack dk

**Pull request merged**